### ISSUE TITLE
Postgres custom `ssl_opts` are not passed to Postgrex config correctly

### ISSUE THREAD

**Issue by jamilbk** [CloudFire-LLC/cloudfire#797]
When we encode the following Ruby hash to JSON, it converts the `ssl_opts`  to a JSON Object:

Ruby:
```rb
default['firezone']['database']['ssl_opts'] = { cacertfile: '/tmp/ca.pem', verify: :verify_none }
```

JSON:
```json
{ "cacertfile": "/tmp/ca.pem", "verify": "verify_none" }
```

Which then turns into this Elixir structure:

```elixir
ssl_opts = %{ "cacertfile" => "/tmp/ca.pem", "verify" => "verify_none" }
```

However, this fails to preserve Atoms which are expected in many ssl configuration parameters.

**Comment by jamilbk** [comment]
@princemaple This seems like it could be an ongoing problem.

We're currently parsing config, mapping all String keys to Atoms [here](https://github.com/firezone/firezone/blob/master/config/runtime.exs#L124):

```elixir
ssl_opts = Keyword.new(database_ssl_opts, fn {k, v} -> {String.to_atom(k), v} end)
parameters = Keyword.new(database_parameters, fn {k, v} -> {String.to_atom(k), v} end)
```

This works for keys, but fails for values that need to be atoms, and other more advanced configuration structures.

Couple ways we could address it:

We could try and detect every string that should be an atom when run through `Jason.decode!`. This would require mapping all possible (or popular?) [`tls_client_option`](https://www.erlang.org/doc/man/ssl.html#type-tls_client_option) atoms from the Erlang SSL module and matching these when a user specifies a matching config item.

Or, since Firezone config is trusted input, we could simple use `Code.eval_string/1` to parse the config and pass it directly into `ssl_opts`. This would be a breaking change when upgrading to `0.5.0` but is likely to be the most flexible option long-term. It saves us from having the Jason.decode! / Jason.encode! in other places for configs as well. All `complex` config could use this facility.

Any thoughts? Doesn't seem to be a *great* solution for this.

**Comment by princemaple** [comment]
> This would require mapping all possible (or popular?) [`tls_client_option`](https://www.erlang.org/doc/man/ssl.html#type-tls_client_option) atoms from the Erlang SSL module and matching these when a user specifies a matching config item.

I'm for this. Mapping popular options - actually only the necessary ones that people explicitly ask for. Majority of the options will never be used.

I don't think it will do well if you ask users to learn the erlang options AND write in elixir.

Currently, how do people what ssl options they can use?

**Comment by princemaple** [comment]
https://github.com/swoosh/swoosh/blob/main/lib/swoosh/adapters/smtp.ex#L61-L103

I did something similar in Swoosh.

This also makes sure that we error properly when given wrong options instead of failing silently... (not sure if erlang ssl yells at you if you provide it with unknown options)

**Comment by jamilbk** [comment]
Fixed by #806
